docs/guide.rst

```diff
@@ -0,0 +1,272 @@
+
+Since :mod:`mozanalysis` is designed around this "standard format", you can pass any of the values in ``res`` to any of the statistics functions, as long as the statistics are suited to the column's type (i.e. binary vs real-valued data)::
+
+    import mozanalysis.bayesian_stats.binary as mabsbin
+    retention_week_2 = mabsbin.compare_branches(res[7], 'active_hours_gt_0')
+
+    import mozanalysis.frequentist_stats.bootstrap as mafsboot
+    boot_uri_week_1 = mafsboot.compare_branches(res[0], 'uri_count', threshold_quantile=0.9999)
+
+    import mozanalysis.bayesian_stats.survival_func as mabssf
+    sf_search_week_2 = mabssf.compare_branches(res[7], 'search_count')
+
+:mod:`dscontrib.flawrence.plot_experiments` has some (shaky) support for visualising stats over time series experiment results.
+
+
+Get the data: cookbook
+=============================
+
+Time series (of analysis windows)
+---------------------------------
+Condensing the above example for simpler copying and pasting::
+
+    dbutils.library.installPyPI("mozanalysis", "[current version]")
+
+    import mozanalysis.metrics.desktop as mmd
+    from mozanalysis.experiment import Experiment
+
+    exp = Experiment(
+        experiment_slug='pref-flip-defaultoncookierestrictions-1506704',
+        start_date='20181217',
+        num_dates_enrollment=8
+    )
+
+    enrollments = exp.get_enrollments(spark).cache()
+
+    res = exp.get_time_series_data(
+        enrollments=enrollments,
+        metric_list=[
+            mmd.active_hours,
+        ],
+        last_date_full_data='20190107',
+        time_series_period='weekly'
+    )
+
+
+One analysis window
+-------------------
+
+If we're only interested in users' (say) second week in the experiment, then we don't need to get a full time series.
+::
+
+    dbutils.library.installPyPI("mozanalysis", "[current version]")
+
+    import mozanalysis.metrics.desktop as mmd
+    from mozanalysis.experiment import Experiment
+
+    exp = Experiment(
+        experiment_slug='pref-flip-defaultoncookierestrictions-1506704',
+        start_date='20181217',
+        num_dates_enrollment=8
+    )
+
+    enrollments = exp.get_enrollments(spark).cache()
+
+    res = exp.get_per_client_data(
+        enrollments=enrollments,
+        metric_list=[
+            mmd.active_hours,
+        ],
+        last_date_full_data='20190107',
+        analysis_start_days=7,
+        analysis_length_days=7
+    )
+
+``last_date_full_data`` is less important for :meth:`mozanalysis.experiment.Experiment.get_per_client_data` than for :meth:`mozanalysis.experiment.Experiment.get_time_series_data`: while ``last_date_full_data`` determines the length of the time series, here it simply sanity checks that the specified analysis window doesn't stretch into the future for any enrolled users.
+
+
+Crunch the stats
```

Comment:
I would prefer to keep the definition of new `Metric` and `DataSource` instances out of this guide, because this guide is meant to be the "happy path"; the first step on the learning curve. If we can reduce the number of concepts that need to be covered by the guide, then it's easier to get started. And hopefully, after the first couple of months, defining new metrics will be rare (~20% of experiments?), so I'd argue that it hopefully doesn't need to be covered by this guide.

I did change the text in L42 to more explicitly refer to the `Metric` docstring - "If not, then you can define your own - see [link to Metric] for examples". Do you think this is enough to tip off users where they need to go to learn how to define metrics and data sources? Or does your judgement differ from mine, and you think it nevertheless needs its own section in this guide (in which case I'll add it without further discussion)?

Reply:
Conciseness is always good. Also, the text does reference the `Metric` docstring as you noted. I believe it is good to leave it as is.